chirp/drivers/tdh8.py

```diff
@@ -0,0 +1,2513 @@
+                    mem.mode = "NFM"
+                    mem.duplex = ""
+                else:
+                    gmrs_freqs = int(GMRS_FREQS_B[mem.number - 1] * 1000000) 
+                    mem.freq = gmrs_freqs
+                    _mem.rxfreq = mem.freq / 10
+                
+            elif float("%i.%06i" % (mem.freq / 1000000, mem.freq % 1000000)) in GMRS_FREQ_A:
+                _mem.rxfreq = mem.freq / 10
+                _mem.offset = 2
+                mem.offset  = 5
+                _mem.txfreq = (mem.freq + 5000000) / 10
+                
+            elif (mem.number >= 55 and mem.number <= 188) and ((mem.freq >= 136000000 and mem.freq <= 174000000) or (mem.freq >= 400000000 and mem.freq <= 520000000)):
+                _mem.rxfreq = mem.freq / 10
+                
+            else:
+                _mem.offset = 0
+                msg = ("The frequency in channels 31-55 must be between 462.55000-462.72500 in 0.025 increments.")
+                raise InvalidValueError(msg)
+                
+        else:
+            # if ((mem.freq >= 136000000 and mem.freq <= 174000000) or (mem.freq >= 400000000 and mem.freq <= 520000000)):
+            #     _mem.rxfreq = _mem.txfreq = mem.freq / 10
+            # else:
+            #     _mem.offset = 0
+            #     msg = ("The frequency must be between 136.00000-174.00000 or 420.00000-450.00000")
+            #     raise InvalidValueError(msg)
+            _mem.rxfreq = _mem.txfreq = mem.freq / 10
+
+
+        if int(self._get_mem(mem.number).rxfreq != 166666665):
+            block_xy = self.get_block_xy(mem.number)
+            _blockfirm = self._get_block_data(block_xy[0])
+            if block_xy[1] == 1:
+                _blockfirm.block1 = 1
+            elif block_xy[1] == 2:
+                _blockfirm.block2 = 1
+            elif block_xy[1] == 3:
+                _blockfirm.block3 = 1
+            elif block_xy[1] == 4:
+                _blockfirm.block4 = 1
+            elif block_xy[1] == 5:
+                _blockfirm.block5 = 1
+            elif block_xy[1] == 6:
+                _blockfirm.block6 = 1
+            elif block_xy[1] == 7:
+                _blockfirm.block7 = 1
+            elif block_xy[1] == 8:
+                _blockfirm.block8 = 1 
+        else:
+            block_xy = self.get_block_xy(mem.number)
+            _blockfirm = self._get_block_data(block_xy[0])
+            if block_xy[1] == 1:
+                _blockfirm.block1 = 0
+            elif block_xy[1] == 2:
+                _blockfirm.block2 = 0
+            elif block_xy[1] == 3:
+                _blockfirm.block3 = 0
+            elif block_xy[1] == 4:
+                _blockfirm.block4 = 0
+            elif block_xy[1] == 5:
+                _blockfirm.block5 = 0
+            elif block_xy[1] == 6:
+                _blockfirm.block6 = 0
+            elif block_xy[1] == 7:
+                _blockfirm.block7 = 0
+            elif block_xy[1] == 8:
+                _blockfirm.block8 = 0
```

Comment:
Are you referring to LN1285 here:
```
 if int(self._get_mem(mem.number).rxfreq != 166666665):
            block_xy = self.get_block_xy(mem.number)
            _blockfirm = self._get_block_data(block_xy[0])
            if block_xy[1] == 1:
                _blockfirm.block1 = 1
```
This is because I found out on the TDH8 protocol that editing the channel frequency needs to set a certain value (blockfirm) to 1, and this channel is considered to be valid for editing. When our channel frequency is empty or in other cases, I will set it Set to 0, set to an invalid edit.

Reply:
I think you're saying that these are "used" or "empty" flags to indicate if the memory has data in it. If so, you should probably be setting these based on `mem.empty`, right? Why would `mem.rxfreq` ever be  `166666665` in `set_memory()`? The user is never going to set it to that - that's the value of when it's unset right?

Either way, I'm saying at least add some comments here to explain what is going on. "blockfirm" doesn't mean much in this context, and it's confusing to read this code. Someone may have to maintain or debug this in the future.